### PR TITLE
doc(termdict) expose structs

### PR DESCRIPTION
**Pull request by saroh**
also add merger doc + lint
refs #1232

### CODE REVIEW

**Review by fulmicoton — APPROVED** [review]

File: src/termdict/mod.rs [hunk 1]
```diff
@@ -20,37 +20,18 @@ as `u64`.
...
-mod tests;
+pub use self::merger::TermMerger;
+pub use self::termdict::TermDictionary;
+pub use self::termdict::TermDictionaryBuilder;
+pub use self::termdict::TermStreamer;
 
 /// Position of the term in the sorted list of terms.
 pub type TermOrdinal = u64;
 
-/// The term dictionary contains all of the terms in
```
Comment (fulmicoton) [inline src/termdict/mod.rs:36]: Do we have comments on the TermDictionary, etc. themselves?

File: src/termdict/merger.rs [hunk 2]
```diff
@@ -39,15 +39,20 @@ impl<'a> TermMerger<'a> {
         }
     }
 
+    /// Iterator over (segment ordinal, [TermOrdinal]) sorted by segment ordinal
+    ///
+    /// This method may be called
+    /// if [Self::advance] has been called before
+    /// and [true] was returned.
```
Comment (saroh) [inline src/termdict/merger.rs]: please check doc for Merger::matching_segments as I wrote the doc without writing the implem and I didn't want to rabbit hole in the code too much ;)

File: src/termdict/mod.rs (hunk 1 above)
Comment (saroh) [reply src/termdict/mod.rs:36]: Yup :) Doc is near duplicated IIRC

File: src/termdict/merger.rs [hunk 3]
```diff
@@ -39,15 +39,20 @@ impl<'a> TermMerger<'a> {
...
     pub fn matching_segments<'b: 'a>(&'b self) -> impl 'b + Iterator<Item = (usize, TermOrdinal)> {
         self.current_segment_and_term_ordinals
             .iter()
             .map(|iv| (iv.index, iv.value))
     }
 
     /// Advance the term iterator to the next term.
-    /// Returns true if there is indeed another term
-    /// False if there is none.
+    /// Returns [true] if there is indeed another term
```
Comment (fulmicoton) [inline src/termdict/merger.rs]: I think the brackets in rustdoc are here to generate links. `true` is what you want if you just want a monospace font.
You can combine both [`TermDictionary::advance`]
```suggestion
    /// Returns `true` if there is indeed another term
```

File: src/termdict/merger.rs [hunk 4]
```diff
@@ -39,15 +39,20 @@ impl<'a> TermMerger<'a> {
...
+    /// [false] if there is none.
```
Comment (fulmicoton) [inline src/termdict/merger.rs]:
```suggestion
    /// `false` if there is none.
```

File: src/termdict/merger.rs (hunk 3 above)
Comment (saroh) [reply src/termdict/merger.rs]: yeah, it's a pity these linked are color coded to the types :)

**Comment by saroh** [comment]
Hi ! I just pushed a few commits for directory and collector packages on this branch : https://github.com/saroh/tantivy/commits/1232-doc-misc would you rather I cherry pick them here or open a new PR ? More may come, it seems to be more linting / fixing links in the other packages :)

**Review by fulmicoton — APPROVED** [review]

**Comment by fulmicoton** [comment]
@saroh. Thanks! I merged this first PR. You can open another one

**Pull request merged**